src_c/rwobject.c

```diff
@@ -435,6 +450,15 @@ _pg_rw_read(SDL_RWops *context, void *ptr, size_t size, size_t maxnum)
     size_t retval;
 #endif /* IS_SDLv2 */
 
+    if (helper->fileno != -1) {
+        retval = read(helper->fileno, ptr, size * maxnum);
+        if (retval == -1) {
+            return -1;
+        }
+        retval /= size;
```

Comment:
I guess it's possible that read could return less than a multiple of size. It's possible read could return 1 byte for example 4 times in a row. I'm not sure how to handle this case really, and it's also quite unlikely.

Reply:
Seems to be true of the other call as well, so it shouldn't be any worse at least.